src/main/java/filter/expression/Qualifier.java

```diff
@@ -527,17 +537,37 @@ private boolean satisfiesUpdatedHours(TurboIssue issue) {
             return false;
         }
 
-        int hoursSinceUpdate;
+        int hoursSinceUpdate = Integer.MIN_VALUE;
 
-        if (issue.getMetadata().isUpdated()) {
-            // Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt.
-            hoursSinceUpdate = Utility.safeLongToInt(issue.getMetadata().getNonSelfUpdatedAt()
-                    .until(getCurrentTime(), ChronoUnit.HOURS));
-        } else {
-            // First time being filtered (haven't gotten metadata from source yet).
-            hoursSinceUpdate = Utility.safeLongToInt(issue.getUpdatedAt().until(getCurrentTime(), ChronoUnit.HOURS));
+        //Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt
+        //and getSelfUpdatedAt
+        switch (updatedKind){
+            case SELF_UPDATED:
+                if (issue.getMetadata().isUpdatedBySelf()) {
+                    // Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt.
+                    hoursSinceUpdate = Utility.safeLongToInt(issue.getMetadata().getSelfUpdatedAt()
+                            .until(getCurrentTime(), ChronoUnit.HOURS));
+                }
+                break;
+            case OTHER_UPDATED:
+                if (issue.getMetadata().isUpdatedByOthers()) {
+                    // Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt.
+                    hoursSinceUpdate = Utility.safeLongToInt(issue.getMetadata().getNonSelfUpdatedAt()
+                            .until(getCurrentTime(), ChronoUnit.HOURS));
+                }
+                break;
+            case ALL_UPDATED:
+                if (issue.getMetadata().isUpdated()) {
+                    // Second time being filtered, we now have metadata from source, so we can use getNonSelfUpdatedAt.
+                    hoursSinceUpdate = Utility.safeLongToInt(issue.getUpdatedAt()
+                            .until(getCurrentTime(), ChronoUnit.HOURS));
+                }
```

Comment:
We should use the conditional to decide the date field to use, instead of the hours. The hours can be computed based on the date field after that.

Also, the comments are duplicated.